### PR TITLE
Setup SVG icon library

### PR DESCRIPTION
**Pull request by nebula-aac**
**Notes for Reviewers**

This PR creates an SVG icon library that consists of most common icons we've been using across multiple repos.

- Cleaned up some of the icons to use props
- Using `SVGAttributes<SVGElement>` to extend
- Commented out some of the icons as they will need to be revised (?)
- Removed hardcoded `fill`, `color`, `stroke`
- Removed conditionals

Revised  
- `TallRoundedRectangleIcon`
- `CircleShapeIcon`

to utilize `filled` and `outlined`

**[Signed commits](../blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: packages/svg/src/icons/types.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export type IconProps = {
+    width?: string | number;
+    height?: string | number;
+    color?: string;
+    fill?: string;
+    style?: CSSProperties | undefined;
+    onClick?: () => void;
+    className?: string;
+}
```
Comment (aabidsofi19) [inline packages/svg/src/icons/types.ts:15]: The Icon Types should inherit from svg ( Our  should be actually just wrappers over svgs )  

```suggestion
import { SVGAttributes } from 'react';
export interface IconProps extends SVGAttributes<SVGElement> {
    children?: never;
    color?: string;
    title?: string;
}
```

File: packages/svg/src/icons/ZoomOutIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+export const ZoomOutIcon: FC<IconProps> = ({
```
Comment (aabidsofi19) [inline packages/svg/src/icons/ZoomOutIcon.tsx:4]: we are still having inconsistency about which properties are supported by different icons .

File: packages/svg/src/icons/ApplicationIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+const ApplicationIcon: FC<IconProps> = ({
```
Comment (aabidsofi19) [inline packages/svg/src/icons/ApplicationIcon.tsx:4]: we should separate solid, outline and  color icons so we dont  have  some icons doing  one thing and others different

File: packages/svg/src/icons/CloudSavedIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+export const CloudSavedIcon: FC<IconProps> = ({
+    width,
+    height,
+    ...props
```
Comment (aabidsofi19) [inline packages/svg/src/icons/CloudSavedIcon.tsx:7]: some icons are supporting additional props but others dont ,  i think every icon should support all the svg attributes .

File: packages/svg/src/icons/CloudSavingIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            viewBox="0 0 19.8 17.5"
+            xmlSpace="preserve"
+            enableBackground="new 0 0 19.8 17.5"
+            width={width}
+            height={height}
+        >
+            <path
+                className="st0"
+                d="M17.9 7c-.2 0-.4-.1-.4-.3C16.4 2.4 12 0 7.8 1.1 5.1 1.9 3 4 2.2 6.7c0 .2-.3.3-.5.2-.2 0-.3-.2-.3-.5C2.7 1.8 7.6-1 12.2.3c3 .8 5.3 3.1 6.1 6.1.1.2-.1.5-.3.5 0 .1 0 .1-.1.1zM9.9 17.5c-3.9 0-7.4-2.6-8.5-6.4-.1-.2.1-.5.3-.5.2-.1.4.1.5.3 1.2 4.2 5.5 6.7 9.8 5.5 2.7-.7 4.8-2.8 5.5-5.5.1-.2.3-.3.5-.3.2.1.3.3.3.5-1 3.8-4.5 6.5-8.4 6.4z"
+                fill="#5f6368"
```
Comment (aabidsofi19) [inline packages/svg/src/icons/CloudSavingIcon.tsx:23]: hardocoded fill ..

File: packages/svg/src/icons/ComponentIcon.tsx [hunk 6]
```diff
@@ -0,0 +1,55 @@
+import { FC } from "react";
+import { IconProps } from "./types";
+
+export const ComponentIcon: FC<IconProps> = ({
```
Comment (aabidsofi19) [inline packages/svg/src/icons/ComponentIcon.tsx:4]: conditionals cause inconsistency

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

**Comment by aabidsofi19** [comment]
@nebula-aac  are we going to publish icons , components etc separately ...

**Comment by aabidsofi19** [comment]
@nebula-aac  i think  we should be fowarding ref to svg , this can be a  beginner issue after  this ..

File: package.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
+    "workspaces": ["./apps/*", "./packages/*"]
```
Comment (aabidsofi19) [inline package.json:2]: i prev had issue with "./packages/*" and had to change to "packages/*"  .

File: packages/svg/src/icons/CloudSavingIcon.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+            width={width}
+            height={height}
+            xmlns="http://www.w3.org/2000/svg"
+            viewBox="0 0 19.8 17.5"
+            xmlSpace="preserve"
+            enableBackground="new 0 0 19.8 17.5"
+        >
+            <path
+                d="M17.9 7c-.2 0-.4-.1-.4-.3C16.4 2.4 12 0 7.8 1.1 5.1 1.9 3 4 2.2 6.7c0 .2-.3.3-.5.2-.2 0-.3-.2-.3-.5C2.7 1.8 7.6-1 12.2.3c3 .8 5.3 3.1 6.1 6.1.1.2-.1.5-.3.5 0 .1 0 .1-.1.1zM9.9 17.5c-3.9 0-7.4-2.6-8.5-6.4-.1-.2.1-.5.3-.5.2-.1.4.1.5.3 1.2 4.2 5.5 6.7 9.8 5.5 2.7-.7 4.8-2.8 5.5-5.5.1-.2.3-.3.5-.3.2.1.3.3.3.5-1 3.8-4.5 6.5-8.4 6.4z"
+                {...props}
```
Comment (aabidsofi19) [inline packages/svg/src/icons/CloudSavingIcon.tsx:16]: we are passing same props to root svg and path  .

**Comment by aabidsofi19** [comment]
Looks great  :)

**Comment by nebula-aac** [comment]
I'm already working on separating the icons so it can either filled or outlined on a separate PR.

**Comment by aabidsofi19** [comment]
@nebula-aac  can you pls create a small readme of how to create a new icon like ( what should be default values and general guidelines  ) .  or maybe a reviewing guide to say

**Pull request merged**